src/composition.rs

```diff
@@ -232,7 +287,13 @@ where
 }
 
 macro_rules! filter {
+    ($values:expr, $operator:expr) => {
+        filter!(None, $values, $operator, exists_op_expr($operator, $values).unwrap())
+    };
     ($key:expr, $values:expr, $operator:expr) => {
+        filter!($key, $values, $operator, binary_op_expr($operator, $key, $values).unwrap())
```

Comment:
1st pattern == matches `exists_op_expr`
2nd pattern == matches `binary_op_expr`

Then those^^ both recursively match to the 3rd filter! pattern.